pkg/kubelet/custommetrics/support.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package custommetrics
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+)
+
+const (
+	// Key for the annotation defining custom metrics for pod.
+	// More info: http://releases.k8s.io/HEAD/docs/proposals/custom-metrics.md#api
+	CustomMetricsAnnotationKey = "metrics.alpha.kubernetes.io/custom-endpoints"
```

Comment:
Hmm. I find that defining API level labels in the Kubelet code to be
strange. How are our users going to identify these labels? Are we hoping
that documentation should solve the discoverability concern?
cc @erictune

On Fri, Dec 11, 2015 at 7:30 AM, Jerzy Szczepkowski <
notifications@github.com> wrote:

> In pkg/kubelet/custommetrics/support.go
> <https://github.com/kubernetes/kubernetes/pull/18295#discussion_r47366605>
> :
>
> > +	"bytes"
> > +	"encoding/json"
> > +	"fmt"
> > +	"io/ioutil"
> > +	"os"
> > +	"path"
> > +
> > +	"github.com/golang/glog"
> > +	"k8s.io/kubernetes/pkg/api"
> > +	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
> > +)
> > +
> > +const (
> > +	// Key for the annotation defining custom metrics for pod.
> > +	// More info: http://releases.k8s.io/HEAD/docs/proposals/custom-metrics.md#api
> > +	CustomMetricsAnnotationKey = "metrics.alpha.kubernetes.io/custom-endpoints"
>
> I thought that annotations keys for alpha features are not kept in the api
> package. Please see:
> https://github.com/kubernetes/kubernetes/blob/master/pkg/client/leaderelection/leaderelection.go#L71
>
> Still, I don't have a strong opinion and can move it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18295/files#r47366605>.
>